packages/@uppy/companion/src/server/provider/dropbox/index.js

```diff
@@ -74,19 +78,31 @@ class DropBox extends Provider {
    */
   async list (options) {
     return this.#withErrorHandling('provider.dropbox.list.error', async () => {
-      const responses = await Promise.all([
-        list(options),
-        userInfo(options),
-      ])
-      // @ts-ignore
-      const [stats, { email }] = responses
+      const userInfoResponse = await userInfo(options)
+      const { email, root_info } = userInfoResponse
+        
+      // Store rootNamespaceId as class member
+      this.rootNamespaceId = root_info?.root_namespace_id
```

Comment:
Just to double check, do we initialize _once_ or per user? Because setting this will only work for the latter.

Reply:
the class is initialised once per *request*, so caching `rootNamespaceId` in `this` doesn't serve any purpose, and should be removed. I think as an optimization we could store `rootNamespaceId` in `providerUserSession`, however we would need to introduce some new logic to be able to do that. see for example how the webdav plugin updates providerUserSession after auth: https://github.com/transloadit/uppy/blob/09048be58b041e725eec670f3535a4f8d4012933/packages/%40uppy/companion/src/server/provider/webdav/index.js#L62
but maybe we don't need that yet.